src/components/shapes/attributes.js

```diff
@@ -169,7 +169,17 @@ module.exports = templatedArray('shape', {
             'See `type` and `xsizemode` for more info.'
         ].join(' ')
     },
-
+    xshift: {
+        valType: 'number',
+        dflt: 0,
+        min: -0.5,
+        max: 0.5,
```

Comment:
We could/should allow ranges beyond [-0.5, 0.5] so that one could highlight a histogram or a bar with width: 1 using a value like 0.7.